include/fe/fe_base.h

```diff
@@ -116,14 +117,22 @@ class FEGenericBase : public FEAbstract
    * Convenient typedefs for gradients of output, hessians of output,
    * and potentially-complex-valued versions of same.
    */
-  typedef OutputType                                                      OutputShape;
-  typedef typename TensorTools::IncrementRank<OutputShape>::type          OutputGradient;
-  typedef typename TensorTools::IncrementRank<OutputGradient>::type       OutputTensor;
-  typedef typename TensorTools::DecrementRank<OutputShape>::type          OutputDivergence;
-  typedef typename TensorTools::MakeNumber<OutputShape>::type             OutputNumber;
-  typedef typename TensorTools::IncrementRank<OutputNumber>::type         OutputNumberGradient;
-  typedef typename TensorTools::IncrementRank<OutputNumberGradient>::type OutputNumberTensor;
-  typedef typename TensorTools::DecrementRank<OutputNumber>::type         OutputNumberDivergence;
+  typedef typename MakeOutput<OutputType>::type                             OutputShape;
```

Comment:
I think that `OutputShape` should be exactly the same as `OutputType`

Reply:
But maybe not ... shape evaluations are a function of chi, eta, etc.... and those *could* be a function of mesh displacement